src/components/autocomplete/js/autocompleteController.js

```diff
@@ -466,9 +466,9 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
         select(ctrl.index);
         break;
       case $mdConstant.KEY_CODE.ESCAPE:
+        event.preventDefault(); // Prevent browser from always clearing input
```

Comment:
I really don't think it should blur if the option is off. I'll have to check it later.

Regardless, I think mdAutocomplete's escape key logic needs work. Also, the input type being search is a forces certain logic instead of being optional.  This isn't tied to just the escape button.

I would argue that input type being search shouldn't be the default and instead it should be text. All of our demos are based around using autocomplete forms, not as a search tool. In fact, we've never written support for the search key which appears on mobile devices. Pressing it on a mdAutocomplete element does nothing and would only confuse users.

Regardless, the change here in this PR is correct, regardless of what we decide to do with the ESCAPE key. We don't need the internal's browser's help to clear text. We handle that ourselves.